horizon/files/horizon_settings/_neutron_settings.py

```diff
@@ -13,6 +13,9 @@
 # The OPENSTACK_NEUTRON_NETWORK settings can be used to enable optional
 # services provided by neutron. Options currenly available are load
 # balancer service, security groups, quotas, VPN service.
+
+{%- set neutron_data = salt['pillar.get']('horizon:server:openstack_neutron_network', {}) %}
```

Comment:
You should reference `app` which is defined in parent template file.
So `app.get('neutron_data', {})`